ENH: Better style handling for component view toggle

TEParsons opened the pull request:
In doing so, introduces a new "HoverMixin" to handle hover events, rather than maintaining basically identical functions in multiple classes.

TEParsons committed:
NF: Move Vector class to its own module, bring up to date with Color

TEParsons committed:
NF: Implement vector at base level

TEParsons committed:
FF: Implement vector in shape

TEParsons committed:
FF: Remove unnecessary Shape size setter

TEParsons committed:
FF: Assurances in vector validation

TEParsons committed:
BF: Grating units should be None not ""

TEParsons committed:
FF: Wrong signatures on getter/setters

TEParsons committed:
FF: Fix dimensions getter

TEParsons committed:
FF: Fix norm setter

TEParsons committed:
FF: Use same name in child classes as parent

TEParsons committed:
FF: Was returing pos rather than size

TEParsons committed:
FF: If given a single value for size, duplicate it

TEParsons committed:
FF: More readable printing for Vectors

TEParsons committed:
FF: Handle None when supplied to size or pos

TEParsons committed:
FF: Return a 1D array if possible, to save extraneous indexing

TEParsons committed:
FF: Use Vector in GratingStim

TEParsons committed:
FF: Use Vector in TextStim

TEParsons committed:
FF: Use Vector in ImageStim

TEParsons committed:
FF: Use Vector in Aperture

TEParsons committed:
FF: Always use float type, even with pix

TEParsons committed:
FF: Remove magnitude setter, not working

TEParsons committed:
DOC: Add documentation to Vector class methods

TEParsons committed:
BF: Integrate size into TextStim, fixes #2381

TEParsons committed:
FF: Handle single size/pos values

TEParsons committed:
FF: Don't bypass setter methods for Aperture pos and size

TEParsons committed:
FF: Integerise pix sizes

TEParsons committed:
FF: Rename internal class in Noise to avoid clashing with Size

TEParsons committed:
NF: Add class to handle vertices, with methods for getting absolute coords relative to an object / a size and pos pair

TEParsons committed:
FF: Subtitute vertices setters for Vertices class

TEParsons committed:
FF: Vertices defaults

TEParsons committed:
FF: Use Size for dotSize, so that it is dynamic with units

TEParsons committed:
FF: Dot size units always pix

TEParsons committed:
FF: Apply changes to _verticesBase to vertices

TEParsons committed:
TEST: scaleFactor no longer needed as size and pos are unit agnostic

TEParsons committed:
FF: Send size argument to grating constructor in RadialStim

TEParsons committed:
FF: Set image size to size of file if None

TEParsons committed:
ENH: Allow width and height setters/getters with all visual objects

TEParsons committed:
ENH: Bring Vertices class in line with Vector, units as properties

TEParsons committed:
ENH: Add flipHoriz and flipVert for convenience

TEParsons committed:
ENH: Move vertices setter up

TEParsons committed:
RF: Do flip calculations in Vertices object rather than .verticesPix

TEParsons committed:
FF: Set pos, posPix and sizePix properly for aperture

TEParsons committed:
RF: Remove unecessary size and width setters from rect

TEParsons committed:
FF: Remove unecessary vertices setters from shape

TEParsons committed:
FF: Set shape vertices properly

TEParsons committed:
FF: Return copy of base vertices rather than handles

TEParsons committed:
FF: norm needs to be doubled/halved

TEParsons committed:
FF: Apply flip to vertices

TEParsons committed:
FF: Image size no longer needs to be doubled

TEParsons committed:
FF: Set vertices before flipping image

TEParsons committed:
FF: Apply flip when creating Vertices objects

TEParsons committed:
FF: Allow Aperture to handle vertices and flip via WindowMixin

TEParsons committed:
FF: Synonymise "None" with "Full Screen"

TEParsons committed:
ENH: Implement size into Textbox2

TEParsons committed:
ENH: Implement letterHeight

TEParsons committed:
ENH: Conveniene function to get letter height in pixels and be guaranteed a single int

TEParsons committed:
RF: Rather than setting correctFlat for the object, use different units

TEParsons committed:
FF: Apply flip before pos, otherwise pos is flipped

TEParsons committed:
FF: Remove duplicate flip setting

TEParsons committed:
FF: Update verts on flip

TEParsons committed:
FF: Use scalefactor in RadialStim test

TEParsons committed:
FF: Only use tessel vertices for base vertices, not border

TEParsons committed:
Merge remote-tracking branch 'upstream/dev' into LayoutClasses

# Conflicts:
#	psychopy/tools/monitorunittools.py
#	psychopy/visual/shape.py

TEParsons committed:
FF: Fix Vector transformation

TEParsons committed:
ENH: Add setter function to Vertices class

TEParsons committed:
ENH: Allow multiplication/division of Vectors by arrays

TEParsons committed:
ENH: Use layout classes for textbox padding

TEParsons committed:
ENH: Use layout classes for textbox letter height

TEParsons committed:
ENH: Add "content box" to Textbox, to keep track of size sans padding, and update size/pos setters to set this

TEParsons committed:
FF: Better None handling in Vectors

TEParsons committed:
ENH: Remove _requestedSize checks and replace with Vector's innate None handling

TEParsons committed:
TEST: New test artefacts for textbox, as previous were inaccurate

TEParsons committed:
TEST: New test artefacts for text, as previous were inaccurate

TEParsons committed:
FF: Update TargetStim to use layout classes

TEParsons committed:
FF: Remove duplicate conversions to allow directly setting shape size and pos as Vector objects

TEParsons committed:
TEST: Change .contains test to use new layout objects rater than doing conversions twice

TEParsons committed:
TEST: Wrong answer for contains test

TEParsons committed:
FF: Set vertices on Line

TEParsons committed:
FF: Set line vertices using start/end rather than None

TEParsons committed:
Merge remote-tracking branch 'upstream/dev' into LayoutClasses

Commit:
FF: Got `degPosFlat` working in layout classes.

Commit:
RF: Removed unused import from test suite.

isolver committed:
BF: pyobjc needs to be < v8.0

BF: pyobjc needs to be < v8.0  or iohub gives error (will investigate later)

Commit:
TESTS: Remove edge condition which fails by modifying the shape.

peircej committed:
Merge pull request #4370 from peircej/release

BF: pyobjc needs to be < v8.0

TEParsons committed:
Merge remote-tracking branch 'upstream/dev' into LayoutClasses

# Conflicts:
#	psychopy/visual/shape.py

TEParsons committed:
Merge remote-tracking branch 'origin/LayoutClasses' into LayoutClasses

peircej committed:
PKG: version bump 2022.1.0rc1

Also added Py2 check to __init__ template

TEParsons committed:
TEST: Add test for color params with dollar

TEParsons committed:
BF: Dollar syntax should override presumption that colors are str if not list

TEParsons committed:
TEST: Make spaces in compiled lists optional for tests

TEParsons committed:
TEST: More informative output for problem experiments test

TEParsons committed:
ENH: Better styling for component panel view toggle buttons

TEParsons committed:
ENH: Better styling for component panel view toggle buttons

TEParsons committed:
ENH: Better styling for component panel view toggle buttons

TEParsons committed:
ENH: Better styling for component panel view toggle buttons

TEParsons committed:
ENH: Make custom toggle button colors settable

TEParsons committed:
ENH: Make custom toggle button colors settable

TEParsons committed:
ENH: Make custom toggle button colors settable

TEParsons committed:
ENH: Make custom toggle button colors settable

TEParsons committed:
ENH: Store hover state for HoverMixin

TEParsons committed:
ENH: Store hover state for HoverMixin

TEParsons committed:
ENH: Store hover state for HoverMixin

TEParsons committed:
ENH: Store hover state for HoverMixin

TEParsons committed:
ENH: Use HoverMixin to handle hover behaviour for component category buttons

TEParsons committed:
ENH: Use HoverMixin to handle hover behaviour for component category buttons

TEParsons committed:
ENH: Use HoverMixin to handle hover behaviour for component category buttons

TEParsons committed:
Merge pull request #4375 from TEParsons/_colorCompile

BF/TEST: Always respect dollar syntax in color params

peircej commented:
Task linked: [CU-1kxnywb Team Science review of component panel toggle buttons](https://app.clickup.com/t/1kxnywb)

isolver committed:
Merge remote-tracking branch 'upstream/release' into merge_release_into_dev

TEParsons committed:
ENH: Print method for vertices objects

peircej committed:
Merge pull request #4382 from isolver/merge_release_into_dev

Merge release into dev to get PR #4375 into dev branch

TEParsons committed:
FF: Flat correction for deg units needs to happen at vertex level, not vector level

TEParsons committed:
Merge pull request #4381 from TEParsons/_optionalSpacesTests

BF: Make spaces optional in problem experiments test

TEParsons committed:
Merge remote-tracking branch 'upstream/dev' into LayoutClasses

Commit:
Merge remote-tracking branch 'origin/_restyleToggle' into _restyleToggle

TEParsons committed:
Merge pull request #4244 from TEParsons/LayoutClasses

NF: Layout classes

TEParsons committed:
ENH: Calculate x coords relative to a midpoint rather than one another

TEParsons committed:
BF: Use vertices object to handle text vertices unit conversion

TEParsons committed:
ENH: Remove now unecessary default size dicts

TEParsons committed:
FF: Scale vertices according to contentBox, not box

TEParsons committed:
FF: Adjust first line Y pos according to ascender, not full height, as full height includes descender but we're only adjusting for ascent

TEParsons committed:
TEST: Update test artefacts as previous were misaligned

TEParsons committed:
Revert "TEST: Update test artefacts as previous were misaligned"

This reverts commit 524dcbcc

TEParsons committed:
FF: Use actual font properties rather than guessing heights

TEParsons committed:
ENH: Add demo to illustrate font layouts

TEParsons committed:
DOC: Clearer explanation of font attrs

TEParsons committed:
RF: Calculate line advance in textbox, not font

TEParsons committed:
TEST: Update test artefacts to new spacing

TEParsons committed:
ENH: Add anchor property to basevisual

TEParsons committed:
ENH: Add anchor property to all shape stim

TEParsons committed:
ENH: Add debug mode to textbox

TEParsons committed:
ENH: Implement anchor for non-editable textbox

TEParsons committed:
ENH: Apply anchor to caret

TEParsons committed:
RF: Handle line spacing via leading in font object

TEParsons committed:
RF: Handle line spacing entirely via leading in font object

TEParsons committed:
DEMO: Update font layout demo

TEParsons committed:
ENH: Implement anchor on aperture

TEParsons committed:
ENH: Implement vertical align bottom

TEParsons committed:
DEMO: Adjust font layout demo to use anchor and alignment

TEParsons committed:
FF: Textbox alignment should default to top left

TEParsons committed:
FF: Implement vertical align center

TEParsons committed:
FF: Caret at 0 should be adjusted by ascender, not height (to ignore leading)

TEParsons committed:
DEMO: Correction to fontLayout demo

TEParsons committed:
ENH: Implement horizontal alignment right

TEParsons committed:
ENH: Implement horizontal alignment center

TEParsons committed:
ENH: Apple horizontal alignment to caret start pos

TEParsons committed:
FF: Apply alignment to uax14 layout too

TEParsons committed:
FF: No longer need to convert line widths from pix units, remove unnecessary double conversion

TEParsons committed:
FF: Layout on setting alignment

TEParsons committed:
FF: Remove print

TEParsons committed:
TEST: Add test for textbox alignment

TEParsons committed:
TEST: Test alignment for each linebreaking style

TEParsons committed:
TEST: Reset textbox params after alignment test

TEParsons committed:
TEST: Specify test alignment and line spacing explicitly

TEParsons committed:
TEST: Update textbox test artefacts

peircej committed:
PKG: set python-vlc versions

On all platforms we need at least 3.0 in order to get the
`CallbackDecorators` attribute.

On Windows the newer versions are reportedly "buggy" so stick with
3.0.11115

https://discourse.psychopy.org/t/moviestim3-1-s-load-time-windows-10-new-problem-in-old-script/23901

peircej committed:
Merge branch 'dev' of https://github.com/psychopy/psychopy into dev

TEParsons committed:
ENH: Replace toggle buttons with a filter dialog

TEParsons committed:
ENH: Replace toggle buttons with a filter dialog

TEParsons committed:
Merge remote-tracking branch 'origin/_restyleToggle' into _restyleToggle

TEParsons committed:
BF: For some reason instructions in keyNameFinder were flipped

TEParsons committed:
ENH: Add alignment param to textbox in builder

TEParsons committed:
TEST: Correct name for test monitor

TEParsons committed:
BF: When setting win, update size and pos object windows

TEParsons committed:
BF: TargetStim should use base win setter

TEParsons committed:
BF: TargetStim should use base win setter

TEParsons committed:
Merge remote-tracking branch 'origin/_textboxLayout' into _textboxLayout

TEParsons committed:
BF: Use win in dict rather than _win

TEParsons committed:
Merge pull request #4387 from TEParsons/_textboxLayout

ENH: Layout textbox with alignment and anchor

TEParsons committed:
Merge remote-tracking branch 'origin/_restyleToggle' into _restyleToggle

TEParsons committed:
ENH: Allow different labels to values in ToggleButtonArray

TEParsons committed:
ENH: More helpful labels in filter buttons

TEParsons committed:
ENH: Make changes live from component filter dlg
